listener.go

```diff
@@ -72,15 +75,26 @@ func (c *Conn) Listener(port uint16, key crypto.PrivateKey) (net.Listener, error
 		return nil, newProtocolError("failed to extract local port")
 	}
 
+	if len(vports) < 1 {
+		return nil, newProtocolError("no virual ports specified")
+	}
+	targetPortStr := strconv.FormatUint((uint64)(tAddr.Port), 10)
+	var portSpecs []OnionPortSpec
+	for _, vport := range vports {
+		portSpecs = append(portSpecs, OnionPortSpec{
+			VirtPort: vport,
+			Target:   targetPortStr,
+		})
+	}
+	cfg.PortSpecs = portSpecs
```

Comment:
Yeah, there are restrictions that apply to PortSpecs: one can map only many-to-one. This is just because target port is the only one per listener. I don't think we should look at the provided port specs here since they are provided as arguments to the listener.